ports/esp8266/machine_uart.c

```diff
@@ -214,47 +211,30 @@ STATIC mp_obj_t pyb_uart_make_new(const mp_obj_type_t *type, size_t n_args, size
     // init the peripheral
     mp_map_t kw_args;
     mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
-    pyb_uart_init_helper(self, n_args - 1, args + 1, &kw_args);
+    mp_machine_uart_init_helper(self, n_args - 1, args + 1, &kw_args);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
-STATIC mp_obj_t pyb_uart_init(size_t n_args, const mp_obj_t *args, mp_map_t *kw_args) {
-    pyb_uart_init_helper(args[0], n_args - 1, args + 1, kw_args);
-    return mp_const_none;
+STATIC void mp_machine_uart_deinit(machine_uart_obj_t *self) {
+    (void)self;
 }
-MP_DEFINE_CONST_FUN_OBJ_KW(pyb_uart_init_obj, 1, pyb_uart_init);
 
-STATIC mp_obj_t pyb_uart_any(mp_obj_t self_in) {
-    pyb_uart_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    return MP_OBJ_NEW_SMALL_INT(uart_rx_any(self->uart_id));
+STATIC mp_int_t mp_machine_uart_any(machine_uart_obj_t *self) {
+    return uart_rx_any(self->uart_id);
 }
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(pyb_uart_any_obj, pyb_uart_any);
 
-STATIC mp_obj_t machine_uart_txdone(mp_obj_t self_in) {
-    pyb_uart_obj_t *self = MP_OBJ_TO_PTR(self_in);
-
-    return uart_txdone(self->uart_id) == true ? mp_const_true : mp_const_false;
+STATIC bool mp_machine_uart_txdone(machine_uart_obj_t *self) {
+    return uart_txdone(self->uart_id);
 }
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_uart_txdone_obj, machine_uart_txdone);
-
-STATIC const mp_rom_map_elem_t pyb_uart_locals_dict_table[] = {
-    { MP_ROM_QSTR(MP_QSTR_init), MP_ROM_PTR(&pyb_uart_init_obj) },
 
-    { MP_ROM_QSTR(MP_QSTR_any), MP_ROM_PTR(&pyb_uart_any_obj) },
-    { MP_ROM_QSTR(MP_QSTR_txdone), MP_ROM_PTR(&machine_uart_txdone_obj) },
-
-    { MP_ROM_QSTR(MP_QSTR_flush), MP_ROM_PTR(&mp_stream_flush_obj) },
-    { MP_ROM_QSTR(MP_QSTR_read), MP_ROM_PTR(&mp_stream_read_obj) },
-    { MP_ROM_QSTR(MP_QSTR_readline), MP_ROM_PTR(&mp_stream_unbuffered_readline_obj) },
-    { MP_ROM_QSTR(MP_QSTR_readinto), MP_ROM_PTR(&mp_stream_readinto_obj) },
-    { MP_ROM_QSTR(MP_QSTR_write), MP_ROM_PTR(&mp_stream_write_obj) },
-};
-
-STATIC MP_DEFINE_CONST_DICT(pyb_uart_locals_dict, pyb_uart_locals_dict_table);
+STATIC void mp_machine_uart_sendbreak(machine_uart_obj_t *self) {
+    (void)self;
+    // TODO
```

Comment:
If I follow correctly, the previous UART object didn't have this function and now it's a no-op? Suggest `mp_raise_NotImplementedError(NULL); // TODO` instead.

Reply:
(After going through all the ports, maybe there's enough ports without `sendbreak` that it's worth making it a config macro flag as well?)

Reply:
OK, I made sendbreak configurable via a macro.